browser/brave_browser_process_impl.cc

```diff
@@ -228,13 +234,13 @@ BraveBrowserProcessImpl::tor_client_updater() {
 }
 #endif
 
-void BraveBrowserProcessImpl::CreateProfileManager() {
-  DCHECK(!created_profile_manager_ && !profile_manager_);
-  created_profile_manager_ = true;
-
-  base::FilePath user_data_dir;
-  base::PathService::Get(chrome::DIR_USER_DATA, &user_data_dir);
-  profile_manager_ = std::make_unique<BraveProfileManager>(user_data_dir);
+brave::BraveP3AService* BraveBrowserProcessImpl::brave_p3a_service() {
+  if (brave_p3a_service_) {
+    return brave_p3a_service_.get();
+  }
+  brave_p3a_service_ = new brave::BraveP3AService(local_state());
+  brave_p3a_service()->InitCallbacks();
```

Comment:
do we need a separate call for this? Can it just happen as part of `new`?

Reply:
Making this a part of the constructor can make testing problematic

Reply:
can we move https://github.com/brave/brave-core/pull/3242/files#diff-ce86d8f829640e5e88ce2b71c4e22b01R63 here then and call `InitCallbacks` inside `Init`? It just seems weird that we're calling `InitCallbacks()`  here and then `Init()` in PreMainMessageLoopRun?

Reply:
I see the comments, but `PreMainMessageLoopRun` is the first time this gets called, right? So it's all happening in the same place anyway?

Reply:
not exactly, this code is eagerly called in `BraveBrowserProcessImpl`. So I think we can't do much about this separation.

Reply:
here? https://github.com/brave/brave-core/pull/3242/files#diff-f4fa6d8248b71f0edf636150386e89ceR88
why are we eagerly calling it there? I think either way we need some comments to help explain

Reply:
here, in constructor https://github.com/brave/brave-core/pull/3242/files#diff-321b7f43759aaadc37d4925c6163cc1bR88
already has a comment

Reply:
why don't we just have a static override for testing? I just saw this again recently and brought me back to this conversation. This doesn't seem like a great way to handle the testing issue imo

Reply:
the comment now says `Can't call it in constructor because of refcounted peculiarities.` - but why is this refcounted in the first place? The lifetime is already tied to BraveBrowserProcessImpl so it will outlive everything

Reply:
well there is a TODO in p3a_service.h saying that we should get rid of recounting :D 
I don't remember the exact reason, probably it is due  tothe fact that histogram events can happen *really* early and *really* late. But maybe just switching it to unique_ptr + Unretained will be fine as is, actualy